### ISSUE TITLE
Illegal Type; shifting seems broken

### ISSUE THREAD

**Issue by Blei** [llvm-dcpu16/llvm-dcpu16#88]
The following short program:

``` c
int shift(int x) {
  return x >> 8;
}
```

results in the following error:

```
clang-3.1: /home/philipp/programming/llvm-dcpu16/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp:702: void {anonymous}::SelectionDAGLegalize::LegalizeOp(llvm::SDNode*): Assertion `TLI.getTypeAction(*DAG.getContext(), Node->getValueType(i)) == TargetLowering::TypeLegal && "Unexpected illegal type!"' failed.
Stack dump:
0.      Program arguments: /home/philipp/programming/llvm-dcpu16-build/bin/clang-3.1 -cc1 -triple dcpu16 -S -disable-free -main-file-name swpb.c -mrelocation-model static -mdisable-fp-elim -mconstructor-aliases -momit-leaf-frame-pointer -coverage-file swpb.s -resource-dir /home/philipp/programming/llvm-dcpu16-build/bin/../lib/clang/3.1 -fmodule-cache-path /var/tmp/clang-module-cache -fno-dwarf-directory-asm -fdebug-compilation-dir /home/philipp/programming/llvm-dcpu16-build -ferror-limit 19 -fmessage-length 87 -mstackrealign -fgnu-runtime -fobjc-runtime-has-arc -fobjc-runtime-has-weak -fobjc-fragile-abi -fdiagnostics-show-option -fcolor-diagnostics -o swpb.s -x c swpb.c 
1.      <eof> parser at end of file
2.      Code generation
3.      Running pass 'Function Pass Manager' on module 'swpb.c'.
4.      Running pass 'DCPU16 DAG->DAG Pattern Instruction Selection' on function '@shift'
clang-3: error: unable to execute command: Aborted
clang-3: error: clang frontend command failed due to signal (use -v to see invocation)
clang-3: note: diagnostic msg: Please submit a bug report to http://llvm.org/bugs/ and include command line arguments and all diagnostic information.
clang-3: note: diagnostic msg: Preprocessed source(s) and associated run script(s) are located at:
clang-3: note: diagnostic msg: /tmp/swpb-f96dcw.i
clang-3: note: diagnostic msg: /tmp/swpb-f96dcw.sh
```

**Comment by ghost** [comment]
Yep, shifting isn't implemented yet. Since I expect the specs of the DCPU16 to be changed at that part very soon, I haven't worked on it till now. Reference to issue #60.

**Comment by ghost** [comment]
Fixed with a04c2fb18907cd221de003f6732cd04be419ba17
